facilitator/src/idl.rs

```diff
@@ -535,6 +461,66 @@ impl Packet for IngestionDataSharePacket {
     }
 }
 
+impl TryFrom<Value> for IngestionDataSharePacket {
+    type Error = Error;
+
+    fn try_from(value: Value) -> Result<Self, Self::Error> {
+        let fields = match value {
+            Value::Record(fields) => fields,
+            _ => {
+                return Err(Error::MalformedDataPacketError(
+                    "value is not a Record".to_owned(),
+                ))
+            }
+        };
```

Comment:
Indeed, that's much more readable -- I replaced all instances of this pattern with something like what you suggest. (I left some `_` matches in the field-by-field extractors, since I can't easily replace those instances with an `if let`.)